assets/section-main-product.css

```diff
@@ -1366,14 +1366,79 @@ a.product__text {
 /* Product-thumbnail snippet */
 
 .product-media-container {
+  --aspect-ratio: var(--preview-ratio);
+  --ratio-percent: calc(1 / var(--aspect-ratio) * 100%);
+  position: relative;
   width: 100%;
   max-width: calc(100% - calc(var(--media-border-width) * 2));
 }
 
+.product-media-container.constrain-height {
+  /* amount to subtract from viewport height when calculating media height */
+  --viewport-offset: 400px;
+  /* smallest height the image is allowed to scale down to */
+  --contrained-min-height: 300px;
+  /* the larger of either the calculated viewport height or the fixed minimum */
+  --constrained-height: max(var(--contrained-min-height), calc(100vh - var(--viewport-offset)));
+  margin-right: auto;
+  margin-left: auto;
+}
+
 @media screen and (min-width: 750px) {
   .product-media-container {
     max-width: 100%;
   }
+
+  .product-media-container:not(.media-type-image) {
+    /* override to use actual media ratio (not poster ratio) for video/models on desktop */
+    --aspect-ratio: var(--ratio);
+  }
+
+  .product-media-container.constrain-height {
+    --viewport-offset: 170px;
+    --contrained-min-height: 500px;
+  }
+
+  .product-media-container.media-fit-cover,
+  .product-media-container.media-fit-cover .product__modal-opener,
+  .product-media-container.media-fit-cover .media {
```

Comment:
I've felt the same way about :is that I do for :where (because they're the same except for the compounding specificity), I don't think the support is _that_ good to use it often (just within latest 2 versions).

However, we could give some thought to using these for styles that wouldn't cause _catastrophic_ layout issues in older browsers. In any case though, to save a line or two I'm not sure it's worth ruining the feature for some users. I think in cases where we'd be avoiding a significant number of repetitive selectors we'd have more reason to use these.

Reply:
yeah I agree. Maybe it can be something we look into once it has a bit more support than the last 2 versions of Safari. 👍